spanner/google/cloud/spanner/snapshot.py

```diff
@@ -81,7 +83,10 @@ def read(self, table, columns, keyset, index='', limit=0,
             transaction=transaction, index=index, limit=limit,
             resume_token=resume_token, options=options)
 
-        return StreamedResultSet(iterator)
+        if self._multi_use:
+            return StreamedResultSet(iterator, source=self)
+        else:
+            return StreamedResultSet(iterator)
 
     def execute_sql(self, sql, params=None, param_types=None, query_mode=None,
                     resume_token=b''):
```

Comment:
If I call `execute_sql` a second time before the first one has returned the transaction id, I will end up creating two transaction ids which is incorrect. You can fix this by either:
- Blocking sending out the second call till you have the transaction id from the first call. But the issue with that is what if the first call for some reason takes a long time to return while the second call would have returned faster.
- Explicitly calling BeginTransaction and blocking sending out any calls till begin transaction has returned. This is the approach we took in Java client to get around the first issue.
In any case you need to block till you have a transaction id.